mbuild/compound.py

```diff
@@ -1009,6 +1010,77 @@ def generate_bonds(self, name_a, name_b, dmin, dmax):
                     self.add_bond((p1, p2))
                     added_bonds.append(bond_tuple)
 
+    @experimental_feature()
+    def freud_generate_bonds(
+        self,
+        name_a,
+        name_b,
+        dmin,
+        dmax,
+        exclude_ii=True,
+    ):
+        """Add Bonds between all pairs of types a/b within [dmin, dmax].
+
+        Parameters
+        ----------
+        name_a : str
+            The name of one of the Particles to be in each bond
+        name_b : str
+            The name of the other Particle to be in each bond
+        dmin : float
+            The minimum distance between Particles for considering a bond
+        dmax : float
+            The maximum distance between Particles for considering a bond
+        exclude_ii : bool, optional, default=True
+            Whether or not to include neighbors with the same index.
+        """
+        freud = import_("freud")
+        if self.box is None:
+            box = self.get_boundingbox()
+        else:
+            box = self.box
+        moved_positions = self.xyz - np.array(
+            [box.Lx / 2, box.Ly / 2, box.Lz / 2]
+        )
+
+        # extend non-periodic dimensions for pseudo-periodicity
+        extended_lengths = list(box.lengths)
+        for i, truthy in enumerate(self.periodicity):
+            if truthy:
+                continue
+            else:
+                extended_lengths[i] = extended_lengths[i] * 10
+        tmp_box = Box(lengths=extended_lengths, angles=list(box.angles))
+        freud_box = freud.box.Box.from_matrix(tmp_box.vectors.T)
+
+        aq = freud.locality.AABBQuery(freud_box, moved_positions)
+
+        # build name_a, name_b mask, other names are -1
+        name_mask = list()
+        name_set = set((name_a, name_b))
+        mask_dict = dict()
+        for i, val in enumerate(name_set):
+            mask_dict[val] = i
+        for part in self.particles():
+            name_mask.append(mask_dict.get(part.name, -1))
+        name_mask = np.array(name_mask)
+
+        nlist = aq.query(
+            moved_positions,
+            dict(
+                r_min=dmin, r_max=dmax, exclude_ii=exclude_ii, num_neighbors=20
```

Comment:
I would remove the `num_neighbors` here -- it will limit the query results if more than 20 neighbors exist in the distance range `r_min, r_max`. Also note that the 20 neighbors it finds could be of any type, which is not good for the filtering call later on because it might hide legitimate A-B bonds. See other comment.
```suggestion
                r_min=dmin, r_max=dmax, exclude_ii=exclude_ii
```